libs/langchain/langchain/storage/file_system.py

```diff
@@ -66,6 +77,21 @@ def _get_full_path(self, key: str) -> Path:
 
         return Path(full_path)
 
+    def _mkdir_for_store(self, dir: Path) -> None:
+        """Makes a store directory path (including parents) with specified permissions
+
+        Args:
+            dir: (Path) The store directory to make
+
+        Returns:
+            None
+        """
+        if not dir.exists():
```

Comment:
```python
In [3]: from pathlib import Path
In [4]: p = Path('foo/bar')
In [5]: p.mkdir?
Signature: p.mkdir(mode=511, parents=False, exist_ok=False)
Docstring: Create a new directory at this given path.
```

Could you rely on pathlib mkdir to simplify the implementation and make it more efficient (the underlying implementation does a similar recursive thing, but with less uneeded calls)

Reply:
@eyurtsev - the reason I could not use pathlib's `mkdir(mode=...)` parameter here is that the `mode` value is restricted by the global `umask` value.

To apply permissions that are not restricted by the `umask` value, they must be applied using an explicit `os.chmod` call.

In fact, I patterned the `_mkdir_for_store()` code after the `Path.mkdir` code, but with the additional `os.chmod()` call.

Do you want me to add a comment to the docstring mentioning this reason why a dedicated directory-making method is needed?

Reply:
@eyurtsev - I added a comment to explain the need for this function.

Reply:
Thank you